src/validation.cpp

```diff
@@ -1740,6 +1745,11 @@ static unsigned int GetBlockScriptFlags(const CBlockIndex* pindex, const Consens
         flags |= SCRIPT_VERIFY_P2SH;
     }
 
+    // Enforce WITNESS rules whenever P2SH is in effect.
```

Comment:
"Enforce WITNESS rules whenever P2SH is in effect and witness deployment is not defined."

... or something

Reply:
I've improved the comments in latest version.